app/code/Magento/Translation/view/base/templates/translate.phtml

```diff
@@ -26,7 +26,7 @@
                 $.initNamespaceStorage('mage-translation-file-version');
                 versionObj = $.localStorage.get('mage-translation-file-version');
 
-                <?php $version = sha1($block->getTranslationFileTimestamp() . $block->getTranslationFilePath()); ?>
+                <?php $version = $block->getTranslationFileVersion(); ?>
 
                 if (versionObj.version !== '<?= /* @noEscape */ $version ?>') {
```

Comment:
Shoudn't we escape version there? Previously it was always sha1, but from now it uses raw value from public method, that could be overridden by plugin.
@ishakhsuvarov i think it should be escaped. what do you think?

Reply:
Sounds completely reasonable to me

Reply:
@ishakhsuvarov pull request prepared https://github.com/magento/magento2/pull/10904